packages/fast-check/src/arbitrary/_internals/implementations/SchedulerImplem.ts

```diff
@@ -184,12 +184,16 @@ export class SchedulerImplem<TMetaData> implements Scheduler<TMetaData> {
           customAct || defaultSchedulerAct,
           () => builder(),
         );
-        scheduled.catch(onFaulty);
         return scheduled;
       };
-      previouslyScheduled = previouslyScheduled.then(onNext);
+      // We will run current item if and only if the one preceeding it succeeds
+      // Otherwise, we mark the run as "failed" and ignore any subsequent item (including this one)
+      previouslyScheduled = previouslyScheduled.then(onNextItem, onFaultyItem);
```

Comment:
The `then` explained:

- success: When item "N-1" is success, we start item "N"
- failure: When item "N-1" (or before) is failure, we mark the sequence as failed and cancel any subsequent item (by throwing)